pkg/handler/sign/hsm_pkcs1_1_5/handler.go

```diff
@@ -0,0 +1,46 @@
+// Package hsm_pkcs1_1_5 provides the HSM based signing method
+// according to PKCS1.1.5.
+package hsm_pkcs1_1_5
+
+import (
+	"crypto"
+	"fmt"
+	"net/http"
+
+	"github.com/miekg/pkcs11"
+
+	"github.com/open-component-model/signing-server/pkg/encoding"
+	"github.com/open-component-model/signing-server/pkg/handler/sign"
+	"github.com/open-component-model/signing-server/pkg/handler/sign/rsassa_pkcs1_1_5"
+	signhttp "github.com/open-component-model/signing-server/pkg/http"
+)
+
+const (
+	// Algorithm defines the type for the RSA PSS signature algorithm.
+	Algorithm = rsassa_pkcs1_1_5.Algorithm
+)
+
+var mechanism = pkcs11.NewMechanism(pkcs11.CKM_RSA_PKCS, nil)
+
+func New(ctx *sign.HSMContext) sign.SignHandler {
+	return &Handler{ctx}
+}
```

Comment:
I like it, because it takes the focus on the meaning and not the implementation, which consists of several fields, which must always belong together and cannot be set independently.

Reply:
I vote for removal of the context object. Just use an Options object that can be passed in and we should be okay.

Reply:
Looking at it again, I dont think we need to support PKCS1_1_5 with HSM, new installations should (and likely will) always use pss. Thus I vote for removal.

Reply:
what is the difference between an Options and a Context object? Just the name?

Why removing it,? It is just a few lines of code here, and provides the same set of functions for HSM and non-HSM.
It really cannot be a matter of maintainability problems?

Reply:
Renamed to HSMOptions and now it is used everywhere as a value object.